homeassistant/components/vesync/sensor.py

```diff
@@ -32,7 +34,7 @@
 class VeSyncSensorEntityDescriptionMixin:
     """Mixin for required keys."""
 
-    value_fn: Callable[[VeSyncDevice], StateType]
+    value_fn: Callable[[VeSyncBaseDevice], StateType]
```

Comment:
`VeSyncBaseDevice` doesn't have a `details` attribute. It's `VeSyncAirBypass`, `VeSyncSwitch` and `VeSyncOutlet` that have a `details` attribute.
```suggestion
    value_fn: Callable[[VeSyncAirBypass | VeSyncOutlet | VeSyncSwitch], StateType]
```